### DISCUSSION TITLE
PseudoBulk differential testing

### DISCUSSION THREAD

**Discussion by Dragonmasterx87** [stuart-lab/signac#1407] [Q&A] [answered]
Hi @timoast sorry for the stupid questions in advance,

So my goal is to perform differential testing on pseudobulk peaks. Therefore I have done the following:

```
combined_processed_atac <- Seurat:::PseudobulkExpression(object = integrated_atac_object, 
                                                         pb.method = 'aggregate', 
                                                         return.seurat = TRUE,
                                                         slot = 'data')
```

The `integrated_atac_object ` contains everything all across the wide-spectrum of your vignettes with no issues with any of them (thank you), the object even contains predicted RNA, because this dataset is where ATAC is performed on similar donors but different cells.

Using the func above I am able to easily perform differential peak testing across pseudobulk'd (pbk'd) data which is segregated on the basis of celltype & sample metadata. All of this is possible, not a problem. 

I am easily able to perform DE testing of pbk'd samples, using DESeq2/LR, to compensate for disproportionate lib size. Now you must be thinking this guy is mad.....if everything is working why are you bothering me...........well...... 😢 

I want to perform DE testing on motifs as well, but on pbk'd data. This isn't possible because the `AddMotifs ` func requires data nested within the `integrated_atac_object@assays[["ATAC"]]@ranges` subslots. From a theoretical perspective, how should I go about this? I don't want to use `LR ` and `latent.vars` on single nuclei data (however I might need to use it on the pbk'd data as you can't perform `latent.vars` with `DESeq2` when testing natively within seurat objcts), as i'm trying to perform sample wide statistical testing. I'd much rather not have to transition to archr, when things are working seamlessly with signac.

Thanks so much for your help.

Cheers,
🐲

**Comment by timoast** [comment, marked as answer]
Hey, for motif enrichment you'd need to input a set of peaks and we test for motif overrepresentation in those sequences compared to background expectation. I think what you want to do here is identify the differential peaks using the pseudobulk data, and then go back to your single-cell object and run the motif enrichment test using those peak ranges as input

**Reply by Dragonmasterx87** [reply]
Thanks for the tips brother. You're the best, have a great day!
